src/main/scala/temple/generate/server/go/service/GoServiceDAOGenerator.scala

```diff
@@ -1,32 +1,179 @@
+    createdByAttribute: Option[CreatedByAttribute],
+    attributes: ListMap[String, Attribute],
+  ): String = {
+    val structFields = ListMap(idAttribute.name.toUpperCase -> generateGoType(idAttribute.attributeType)) ++
+      createdByAttribute.map(createdByAttribute =>
+        createdByAttribute.name.capitalize -> generateGoType(createdByAttribute.attributeType),
+      ) ++
+      attributes.map { case (name, attribute) => (name.capitalize -> generateGoType(attribute.attributeType)) }
 
-  private[go] def generateStructs(): String =
     mkCode.lines(
-      "// DAO encapsulates access to the database",
-      s"type DAO struct ${CodeWrap.curly.tabbed("DB *sql.DB")}",
+      s"// ${serviceName.capitalize} encapsulates the object stored in the datastore",
+      mkCode(
+        s"type ${serviceName.capitalize} struct",
+        CodeWrap.curly.tabbed(
+          CodeUtils.pad(structFields),
+        ),
+      ),
+    )
+  }
+
+  private def generateInputStructCommentSubstring(operation: CRUD, serviceName: String): String =
+    operation match {
+      case ReadAll                         => s"read a $serviceName list"
+      case Create | Read | Update | Delete => s"${verb(operation).toLowerCase()} a single $serviceName"
+    }
+
+  private def generateInputStruct(
+    serviceName: String,
+    operation: CRUD,
+    idAttribute: IDAttribute,
+    createdByAttribute: Option[CreatedByAttribute],
+    attributes: ListMap[String, Attribute],
+  ): String = {
+    val structName = s"${generateDAOFunctionName(operation, serviceName)}Input"
+    mkCode.lines(
+      s"// $structName encapsulates the information required to ${generateInputStructCommentSubstring(operation, serviceName)} in the datastore",
+      mkCode(
+        s"type $structName struct",
+        CodeWrap.curly.tabbed(
+          CodeUtils.pad(
+            operation match {
+              case ReadAll =>
+                val createdByAttributeVal = createdByAttribute.getOrElse {
+                  throw new Exception(
+                    "Programmer error: createdByAttribute should be present because input struct for readAll " +
+                    "operation should only be generated if enumerating by creator",
+                  )
+                }
+                ListMap(
+                  createdByAttributeVal.inputName.capitalize -> generateGoType(createdByAttributeVal.attributeType),
+                )
+              case Create =>
+                ListMap(idAttribute.name.toUpperCase -> generateGoType(idAttribute.attributeType)) ++
+                createdByAttribute.map(createdByAttribute =>
+                  createdByAttribute.inputName.capitalize -> generateGoType(createdByAttribute.attributeType),
+                ) ++
+                attributes.collect {
+                  case (name, attribute) if attribute.accessAnnotation != Some(Annotation.ServerSet) =>
+                    (name.capitalize, generateGoType(attribute.attributeType))
+                }
```

Comment:
This looks very similar to the code on 87-91. Can we put the common parts in a function with an understandable name? Might make it easier to see what's going on